translate refs-must-have-owner warning (#112)
reactjs/de.reactjs.org: content/warnings/refs-must-have-owner.md

permalink: warnings/refs-must-have-owner.html
---

Du bist wahrscheinlich hier, weil eine der folgenden Fehlermeldungen angezeigt wird:

*React 16.0.0+*
> Warning:
>
> Element ref was specified as a string (myRefName) but no owner was set. You may have multiple copies of React loaded. (details: https://fb.me/react-refs-must-have-owner).

*frühere Versionen von React*
> Warning:
>
> addComponentAsRefTo(...): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component's `render` method, or you have multiple copies of React loaded.

Dies bedeutet normalerweise eines von drei Dingen:

- Du versuchst, einer Funktionskomponente ein `ref` hinzuzufügen.
- Du versuchst, einem Element, das außerhalb der render() Funktion einer Komponente erstellt wird, ein `ref` hinzuzufügen.
- Du hast mehrere (in Konflikt stehende) Kopien von React geladen (z. B. aufgrund einer falsch konfigurierten npm-Abhängigkeit)

## Refs auf Funktionskomponenten {#refs-on-function-components}

> Wenn `<Foo>` eine Funktionskomponente ist, kannst du keine `ref` hinzufügen:

Reviewer: In Line 27 you're using the female version of `ref` => `keine`.
In Line 21 you're using the male version of `ref` => `ein`.

Please choose one and double-check if you actually use the chosen version in all occurencies in this file :+1:

```js

> // Doesn't work if Foo is a function!

Reviewer 2: ```suggestion
// Funktioniert nicht, wenn Foo eine Funktion ist!
```

<Foo ref={foo} />
```

> Wenn du einer Komponente einen `ref` hinzufügen musst, konvertierst du ihn zuerst in eine Klasse, oder verwendest du keine `ref`, da diese [selten erforderlich](/docs/refs-and-the-dom.html#when-to-use-refs) sind.

Reviewer: ```suggestion
Wenn du einer Komponente einen `ref` hinzufügen musst, konvertiere ihn zuerst in eine Klasse, oder verwende keine `ref`, da diese [selten erforderlich](/docs/refs-and-the-dom.html#when-to-use-refs) sind.
```

Feels like `konvertiere ihn` and `verwende keine` sounds more friendly than `konvertierst du` oder `verwendest du`.

Something like giving `suggestions` vs. `instructions`.

> ## Zeichenfolgenreferenzen außerhalb der Rendermethode {#strings-refs-outside-the-render-method}

Reviewer: `Zeichenfolgenreferenzen` sounds really weird,
especially because I have the feeling,
that a beginner would not understand what this is.

Does anyone actually use `Zeichenfolge` in real life,
instead of `string`?

> Dies bedeutet normalerweise, dass du versuchst, einer Komponente, die keinen Eigentümer hat (d. h. nicht innerhalb der `render` Methode einer anderen Komponente erstellt wurde), einen `ref` hinzuzufügen. Zum Beispiel, das funktioniert nicht:

Reviewer 2: ```suggestion
Dies bedeutet normalerweise, dass du versuchst, einer Komponente, die keinen Eigentümer hat (d. h. nicht innerhalb der `render` Methode einer anderen Komponente erstellt wurde), einen `ref` hinzuzufügen. Zum Beispiel funktioniert das nicht:
```

> Dies bedeutet normalerweise, dass du versuchst, einer Komponente, die keinen Eigentümer hat (dh nicht innerhalb der `render` Methode einer anderen Komponente erstellt wurde), einen `ref` hinzuzufügen. Zum Beispiel, dies funktioniert nicht:

Reviewer: ```suggestion
Dies bedeutet normalerweise, dass du versuchst, einer Komponente, die keinen Eigentümer hat (d. h. nicht innerhalb der `render` Methode einer anderen Komponente erstellt wurde), einen `ref` hinzuzufügen. Zum Beispiel, das funktioniert nicht:
```

[d. h.](https://de.wiktionary.org/wiki/d._h.)

`das` sounds more casual than `dies`.

```js

> // Doesn't work!

Reviewer 2: ```suggestion
// Funktioniert nicht!
```

ReactDOM.render(<App ref="app" />, el);
```

> Versuche, diese Komponente in einer neuen Komponente der obersten Ebene zu rendern, die den `ref` enthält. Alternativ kannst du einen callback ref verwenden:

Reviewer 2: ```suggestion
Versuche, diese Komponente in eine neue Top-Level-Komponente zu rendern, die den `ref` enthält. Alternativ kannst du einen callback ref verwenden:
```
Maybe we should leave "Top-Level" as it is.

> Versuchst du, diese Komponente in einer neuen Komponente der obersten Ebene zu rendern, die den `ref` enthält. Alternativ kannst du einen Rückruf `ref` verwenden:

Reviewer: ```suggestion
Versuche, diese Komponente in einer neuen Komponente der obersten Ebene zu rendern, die den `ref` enthält. Alternativ kannst du einen callback ref verwenden:
```

`Rückruf ref` sounds really strange.
I think users, who ever need this construction, know what a callback is.

);
```

> Überlegst du, ob du [wirklich eine Referenz benötigst](/docs/refs-and-the-dom.html#when-to-use-refs), bevor du diesen Ansatz verwendest.

Reviewer: ```suggestion
Überlege, ob du [wirklich eine Referenz benötigst](/docs/refs-and-the-dom.html#when-to-use-refs), bevor du diesen Ansatz verwendest.
```

## Mehrere Kopien von React {#multiple-copies-of-react}

> Bower leistet gute Arbeit bei der Deduplizierung von Abhängigkeiten, npm jedoch nicht. Wenn du mit `ref` nichts (Lustiges) machst, liegt das Problem wahrscheinlich nicht an deinen `ref`, sondern eher daran, dass mehrere Kopien von React in dein Projekt geladen wurden. Wenn du ein Modul eines Drittanbieters über npm aufrufst, erhaltest du manchmal eine doppelte Kopie der Abhängigkeitsbibliothek, was zu Problemen führen kann.

Reviewer: ```suggestion
Bower leistet gute Arbeit bei der Deduplizierung von Abhängigkeiten, npm jedoch nicht. Wenn du mit `ref` nichts (Ausgefallenes) machst, liegt das Problem wahrscheinlich nicht an deinen `ref`, sondern eher daran, dass mehrere Kopien von React in dein Projekt geladen wurden. Wenn du ein Modul eines Drittanbieters über npm aufrufst, erhältst du manchmal eine doppelte Kopie der Abhängigkeitsbibliothek, was zu Problemen führen kann.
```

[erhalten](https://de.wiktionary.org/wiki/erhalten)

> Bower leistet gute Arbeit bei der Deduplizierung von Abhängigkeiten, npm jedoch nicht. Wenn du mit `ref` nichts (Ausgefallenes) machst, liegt das Problem wahrscheinlich nicht an deinen `ref`, sondern eher daran, dass mehrere Kopien von React in dein Projekt geladen wurden. Wenn du ein Modul eines Drittanbieters über npm aufrufst, erhältst du manchmal eine doppelte Kopie der Abhängigkeitsbibliothek, was zu Problemen führen kann.

Reviewer 2: ```suggestion
Bower leistet gute Arbeit bei der Deduplizierung von Abhängigkeiten, npm jedoch nicht. Wenn du mit `ref` nichts (Ausgefallenes) machst, liegt das Problem wahrscheinlich nicht an deinen Refs, sondern eher daran, dass mehrere Kopien von React in dein Projekt geladen wurden. Wenn du ein Modul eines Drittanbieters über npm aufrufst, erhältst du manchmal eine doppelte Kopie der Abhängigkeitsbibliothek, was zu Problemen führen kann.
```

Author: Why change only one of the two back to plural?

Reviewer 2: Updated it.

> Wenn du npm verwendest, kann `npm ls` oder `npm ls react` zum Aufleuchten beitragen.

Reviewer: ```suggestion
Wenn du npm verwendest, kann `npm ls` oder `npm ls react` zum Aufklären beitragen.
```
